salty-747/html_ui/Pages/VCockpit/Instruments/Airliners/B747_8/FMC/B747_8_FMC_ProgPage.js

```diff
@@ -145,22 +148,59 @@ class B747_8_FMC_ProgPage {
         else {
             crzSpeedCell = Simplane.getAutoPilotAirspeedHoldValue().toFixed(0);
         }
+        let todTimeCell = "";
+        let todDistanceCell = "";
+        let toToDText = "";
+        let distanceToTOD = SimVar.GetSimVarValue("L:WT_CJ4_TOD_REMAINING", "number");
+        if (distanceToTOD > 0) {
+            if (isFinite(distanceToTOD) && distanceToTOD < 201) {
+                toToDText = "TO T/D";
+                if (distanceToTOD < 5) {
+                    todDistanceCell = "NOW";
+                } else {
+                    todDistanceCell = distanceToTOD.toFixed(0)  + "NM";
+                    todTimeCell = "&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp";
+                    let eta = undefined;
+                    eta = (B747_8_FMC_ProgPage.computeEtaToWaypoint(distanceToTOD, speed) + currentTime) % 86400;
+                    if (isFinite(eta)) {
+                        let etaHours = Math.floor(eta / 3600);
+                        let etaMinutes = Math.floor((eta - etaHours * 3600) / 60);
+                        todTimeCell += etaHours.toFixed(0).padStart(2, '0') + etaMinutes.toFixed(0).padStart(2, '0') + "z"+"&nbsp"+"/";
+                    } else {
+                        todTimeCell += "&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp";
+                    }
+                }
+            }
+            else {
+                toToDText = "";
+                todTimeCell = "";
+                todDistanceCell = "";
+            }
+        }
         fmc.setTemplate([
             [progressTitle],
             ["\xa0TO", "FUEL", "DTG\xa0\xa0ETA"],
             [waypointActiveCell + "[color]magenta", waypointActiveFuelCell, waypointActiveDistanceCell],
-            ["\xa0NEXT", "", "\xa0\xa0\xa0\xa0\xa0ETA"],
+            ["\xa0NEXT", "", ""],
             [waypointActiveNextCell, waypointActiveNextFuelCell, waypointActiveNextDistanceCell],
             ["\xa0DEST"],
             [destinationCell, destinationFuelCell, destinationDistanceCell],
-            ["\xa0SEL SPD"],
-            [crzSpeedCell],
-            [],
-            [""],
+            ["\xa0SEL SPD", toToDText, ""],
+            [crzSpeedCell + "[color]magenta", todDistanceCell, todTimeCell],
+            ["", "", ""],
+            ["", "", ""],
             ["__FMCSEPARATOR"],
             ["<POS REPORT", "POS REF>"]
         ]);
     }
+    static computeEtaToWaypoint(distance, groundSpeed) {
+        if (groundSpeed < 50) {
+            groundSpeed = 50;
+        }
+        if (groundSpeed > 0.1) {
+            return distance / groundSpeed * 3600;
+        }
+    }
```

Comment:
preferably do not overwrite arguments passed in. also, the second if is not needed if the first one is going to always set the ground speed to 50 if it's less than 50. is it actually supposed to be set to 50 if it's less?